sdk/highlightinc-highlight-datasource/src/components/VariableQueryEditor.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useState } from 'react';
+import { SelectableValue } from '@grafana/data';
+import { InlineField, InlineFieldRow, Input, Select } from '@grafana/ui';
+
+import { tableOptions } from '../datasource';
+import { HighlightVariableQuery, Table } from '../types';
+
+interface VariableQueryProps {
+  query: HighlightVariableQuery;
+  onChange: (query: HighlightVariableQuery) => void;
+}
+
+export const VariableQueryEditor = ({ onChange, query }: VariableQueryProps) => {
+  const { key, table } = query;
+  const [keyText, setKeytext] = useState<string | undefined>(key);
+
+  const handleTableChange = (option: SelectableValue<Table>) => {
+    const newTable = option.value;
+    onChange({ ...query, table: newTable });
+  };
+
+  const handleKeyChange = (option: SelectableValue<string>) => {
+    setKeytext(option.target.value);
+  };
+
+  const handleKeyBlur = () => {
+    onChange({ ...query, key: keyText });
+  };
+
+  return (
+    <>
+      <InlineFieldRow>
+        <InlineField label="Table" labelWidth={10}>
```

Comment:
Nit but might be nice to call this "resource" to match its name in the query editor